sections/producer.tex

```diff
@@ -1,5 +1,26 @@
+The \CCSPStlye{Producer} class is used to calculate higher-level variables and define new array columns to be written o disk. The corresponding task is called \CCSPStlye{cf.ProduceColumns} (see Ref.~\cite{cf_repo} for detailed info). Naturally, we only want to compute these new variables for the relevant events for our analysis. Thus, the producers are executed after the selection step in the task graph. 
 
-In this part of the tutorial, we will write a producer which calculates the four lepton invariant mass.
+ColumnFlow also provides \CCSPStlye{Producer}s to compute commonly used event information, such as MC, pdf or pileup weights. These can be found under \code{columnflow.production.cms}. The \code{H4L} analysis includes three exemplary \CCSPStlye{Producer}s in \code{h4l/production/example.py}. You will notice that the script starts by importing all relevant modules, including CMS specific ones. We also load both \code{numpy} and \code{awkward} with the \code{maybe\_import} mechanism.
+
+We start by defining a new \CCSPStlye{Producer} class named \code{features}. This class requires the transverse jet momentum \code{Jet.pt}, which must be added to its \code{uses} set, and produces two new array columns, the total jet transverse momentum \code{ht} and the number of jets in an event \code{n\_jet}, which are both added to its \code{produces} set. Each of these new variables is computed and stored to disk using the \code{set\_ak\_column} function. Note that for the case of \code{n\_jet}, we specified that the column element must be an \code{int} value.
+
+The second \CCSPStlye{Producer} class \code{cutflow\_features} allows us to define and store features to be used for cutflow plots. Here, in addition to \code{Jet.pt} we also require \code{mc\_weight} and \code{category\_ids} to be added to the \code{uses} set. Note that both of these are \CCSPStlye{Producer}s themselves which you can find by following the import path at the beginning of the script. 
+
+The  \CCSPStlye{Producer} class \code{mc\_weight} reads in the \code{genWeight} column and, if existent, the \code{LHEWeight} column, both stored in \code{events}. Since these columns are required, they are both added to the \code{uses} set of \code{mc\_weight}. By extension, when we call \code{mc\_weight} in our \code{uses} set, we are calling these columns as well. The \code{mc\_weight} class simply decides which one of these weights to use and saves the decision as a new column, also named \code{mc\_weight}, which is included in its \code{produces} set. Again, by extension, we must add the \code{mc\_weight} class to our own \code{produces} set such that this new column also gets created and saved to disk. 
```

Comment:
Regarding the last sentence: `must` is a very strong word, and in principle we also don't have to write the `mc_weight` column here just because we call the producer here. In principle, we could also call the producer in multiple other producers, but not all of them would have to save this column right? I'm worried that users might interpret this such that they _must_ include all external producers to both the `uses` and `produces` set or otherwise break the code. On the other hand, what you write is correct of course. To write the information to disk, we have to specify the producer in `produces`. I'm not sure how to do this better

Reply:
I see what you mean. I can change it to `At this point, we also have the option to add the \code{mc\_weight} class to our own \code{produces} set. In this way, the new column also gets created and saved to disk. `